### ISSUE TITLE
Refactor methods in BgeeDBUtils

### ISSUE THREAD

**Issue by fbastian** [BgeeDB/bgee_apps#8]
In GitLab by @fbastian on Jul 14, 2015, 13:31

Lots of methods in `org.bgee.pipeline.BgeeDBUtils` could be refactored. For instance, all the `getXXXTOsByIds` could use a common method accepting as argument a `DAOResultSet<? extends TransferObject>`. 

See method `org.bgee.pipeline.BgeeDBUtils.generateNamesByIdsMap` for a similar approach.

**Comment by fbastian** [comment]
In GitLab by @vrechdelaval on Jul 15, 2015, 19:38

mentioned in commit d3bd31cd2f0d675f0106b84cf064626dd4586dac

**Comment by fbastian** [comment]
In GitLab by @fbastian on Jul 16, 2015, 12:26

Or a `DAOResultSet<? extends EntityTO>` :p

**Comment by fbastian** [comment]
In GitLab by @vrechdelaval on Jul 20, 2015, 12:40

Is there any reason to use `throw log.throwing(new AssertionError("<message>")` instead of `org.junit.Assert.fail("<message>")` in tests?

**Comment by fbastian** [comment]
In GitLab by @vrechdelaval on Jul 20, 2015, 15:35

mentioned in commit 6829afdd91d3c1b14834d37d3452cab558267c67

**Comment by fbastian** [comment]
In GitLab by @vrechdelaval on Jul 20, 2015, 15:46

See `git diff 66ad02e3..2329b6d3`

**Comment by fbastian** [comment]
In GitLab by @fbastian on Aug 24, 2015, 13:20

* line 337 `T entityTO = rs.getTO();`: won't you have a `ClassCastException` if the `ResultSet` does not return an `EntityTO`?
* method `generateNamesByIdsMap`: wouldn't it be simpler for this method to call `generateTOsByIdsMap`, and then to iterate the returned `Map` to generate a new one, with the mapping ID -> name?
* Why do we need the methods `getCIOStatementTOsByIds`, or `getGeneTOsByIds`, rather than using directly `generateTOsByIdsMap`. 
* Same question for, e.g., `getGeneNamesByIds`
* In the issue description, it is written "*For instance*, all the getXXXTOsByIds could use a common method": is there any other refactoring possible?

**Comment by fbastian** [comment]
In GitLab by @vrechdelaval on Aug 24, 2015, 20:33

* We don't need `ClassCastException` because `T extends EntityTO`.
* I added call to `generateTOsByIdsMap` in `generateNamesByIdsMap` method (see commit 141dfad5).
* Currently, we keep specific methods to retrieve names by IDs to be able to store and restore specific attributes. This could be generic in generateNamesByIdsMap using Java 8, but we are just lazy to implement it now. I added comments (see commit 7e36b203)
* We can use directly generateTOsByIdsMap to retrieve TOs by IDs but why don't keep the same schema to retrieve names by IDs?
* I don't see any other possible refactoring in this class.
